src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -2642,8 +2643,403 @@ public override string TypeNameOfValue
+                case 2: return typeof(PSGenericValueType2);
+                case 3: return typeof(PSGenericValueType3);
+                case 4: return typeof(PSGenericValueType4);
+                case 5: return typeof(PSGenericValueType5);
+                case 6: return typeof(PSGenericValueType6);
+                case 7: return typeof(PSGenericValueType7);
+                case 8: return typeof(PSGenericValueType8);
+                case 9: return typeof(PSGenericValueType9);
+                case 10: return typeof(PSGenericValueType10);
+                case 11: return typeof(PSGenericValueType11);
+                case 12: return typeof(PSGenericValueType12);
+                case 13: return typeof(PSGenericValueType13);
+                case 14: return typeof(PSGenericValueType14);
+                case 15: return typeof(PSGenericValueType15);
+                case 16: return typeof(PSGenericValueType16);
+                default:
+                    return typeof(PSGenericValueType<>).MakeGenericType(GetGenericType(i - 1));
+            }
+        }
+    }
+
+    struct PSGenericValueType0 { internal int value; }
+    struct PSGenericValueType1 { internal int value; }
+    struct PSGenericValueType2 { internal int value; }
+    struct PSGenericValueType3 { internal int value; }
+    struct PSGenericValueType4 { internal int value; }
+    struct PSGenericValueType5 { internal int value; }
+    struct PSGenericValueType6 { internal int value; }
+    struct PSGenericValueType7 { internal int value; }
+    struct PSGenericValueType8 { internal int value; }
+    struct PSGenericValueType9 { internal int value; }
+    struct PSGenericValueType10 { internal int value; }
+    struct PSGenericValueType11 { internal int value; }
+    struct PSGenericValueType12 { internal int value; }
+    struct PSGenericValueType13 { internal int value; }
+    struct PSGenericValueType14 { internal int value; }
+    struct PSGenericValueType15 { internal int value; }
+    struct PSGenericValueType16 { internal int value; }
+
+    struct PSGenericValueType<T> { internal int value; }
+
+    struct PSEnum<T> { }
+
+    struct PSPointer<T> { }
+
+    struct PSTypedReference { }
+
+    internal abstract class MethodGroup { }
+    internal class MethodGroup<T1> : MethodGroup { }
+    internal class MethodGroup<T1, T2> : MethodGroup { }
+    internal class MethodGroup<T1, T2, T3, T4> : MethodGroup { }
+    internal class MethodGroup<T1, T2, T3, T4, T5, T6, T7, T8> : MethodGroup { }
+    internal class MethodGroup<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16> : MethodGroup { }
+    internal class MethodGroup<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16, T17, T18, T19, T20, T21, T22, T23, T24, T25, T26, T27, T28, T29, T30, T31, T32> : MethodGroup { }
+
+    class Unit
+    {
+        private Unit() { }
+    }
+
+    internal struct PSMethodSignatureEnumerator : IEnumerator<Type>
+    {
+        private int _current;
```

Comment:
`_current` is not the backing field for `Current`, you should rename `_current` to avoid any confusion.

Reply:
good point